Allow overriding initial peering DNS by environment variable

dsiganos opened the pull request:
The initial peering DNS is peering.nano.org.
This commit allows to set NANO_DEFAULT_PEER to override it.
This allows for more flexibility when running private test networks.

dsiganos committed:
Allow overriding initial DNS of beta network by environment variable

The initial peering DNS for beta network is peering-beta.nano.org.
This commit allows to set NANO_BETA_PEER_NETWORK to override it.
This allows for more flexibility when running private test networks.

zhyatt commented:
@dsiganos In what scenario is using the beta network with custom peering DNS preferred over the test network which already allows this customization?

dsiganos commented:
If we wanted to run SPAM attack tests then they are easier on beta network than on test network due to reduced difficulty.
Additionally, being able to override the peering DNS to something that doesn't resolve, is a very easy way to stop a node from contacting any peers (after wiping the peer table).

dsiganos commented:
After discussion, it was decided to:
* Rename NANO_TEST_PEER_NETWORK to NANO_DEFAULT_PEER
* Apply it to the active network (live, beta, test)

dsiganos committed:
Rename NANO_TEST_PEER_NETWORK to NANO_DEFAULT_PEER and apply it to all networks (live, beta, test)

We want to have one environment variable to override the default
preconfigured peer in all networks cases (live, beta and test).

The name thus needs renaming to remove TEST from its name.
Since we are renaming it, we also take the opportunity to improve the name
to something more descriptive, NANO_DEFAULT_PEER.

zhyatt commented:
Just noting that there are no documentation changes to be considered for this.
